backend/substrapp/views/model.py

```diff
@@ -116,6 +118,19 @@ def details(self, request, *args, **kwargs):
         return Response(data, status=status.HTTP_200_OK)
 
 
+def gzip_action(func):
+    gz = GZipMiddleware()
+
+    @wraps(func)
+    def wrapper(self, request, *args, **kwargs):
+        resp = func(self, request, *args, **kwargs)
+        if request.method == 'GET':
```

Comment:
Only in the case of response with status code `200` will return a model content, in other case if will fail.
Maybe you can add something like this : `if request.method == 'GET' and resp.status_code == 200:`

Reply:
I don't think it's necessary to add a condition because even if the request doesn't respond with `200`, compressing the response body doesn't hurt. The middleware won't compress short payloads anyway (<200 bytes), so "Page Not Found" kind-of-responses won't be compressed

We could even remove the condition `if request.method == 'GET'`, actually!